### PR TITLE
PRIME-1398 Send Email (Bulk Actions)

### PR DESCRIPTION
**Pull request by AnaisHebert**

### CODE REVIEW

File: prime-dotnet-webapi/Controllers/EnrolleesController.cs [hunk 1]
```diff
@@ -862,5 +862,24 @@ public async Task<ActionResult<EnrolleeNoteViewModel>> DeleteEnrolleeNotificatio
...
             return Ok();
         }
+
+        // GET: api/Enrollees/emails
+        /// <summary>
+        /// Gets all of the enrollee emails that match the type
+        /// </summary>
+        /// <param name="bulkEmailType"></param>
+        [HttpGet("emails", Name = nameof(GetEnrolleeEmails))]
+        [ProducesResponseType(typeof(ApiBadRequestResponse), StatusCodes.Status400BadRequest)]
```
Comment (james-hollinger) [inline prime-dotnet-webapi/Controllers/EnrolleesController.cs:872]: This endpoint doesn't return status code 400, but it does implicitly return 403.

**Review by mtpultz — APPROVED** [review]
image

File: prime-angular-frontend/src/app/shared/components/dialogs/content/send-bulk-email/send-bulk-email.component.spec.ts [hunk 2]
```diff
@@ -0,0 +1,25 @@
...
+  });
+
+  beforeEach(() => {
+    fixture = TestBed.createComponent(SendBulkEmailComponent);
+    component = fixture.componentInstance;
+    fixture.detectChanges();
+  });
+
+  it('should create', () => {
+    expect(component).toBeTruthy();
```
Comment (johnnyl6) [inline prime-angular-frontend/src/app/shared/components/dialogs/content/send-bulk-email/send-bulk-email.component.spec.ts:23]: Test is currently broken due to missing DI provider.

```
SendBulkEmailComponent > should create

NullInjectorError: R3InjectorError(DynamicTestModule)[InjectionToken MatDialogData -> InjectionToken MatDialogData]: 
  NullInjectorError: No provider for InjectionToken MatDialogData!
error properties: Object({ ngTempTokenPath: null, ngTokenPath: [ 'InjectionToken MatDialogData', 'InjectionToken MatDialogData' ] })
NullInjectorError: R3InjectorError(DynamicTestModule)[InjectionToken MatDialogData -> InjectionToken MatDialogData]: 
  NullInjectorError: No provider for InjectionToken MatDialogData!
```

File: prime-dotnet-webapi/Services/EnrolleeService.cs [hunk 3]
```diff
@@ -903,5 +904,28 @@ public async Task<IEnumerable<int>> GetNotifiedEnrolleeIdsForAdminAsync(ClaimsPr
...
                 .ToListAsync();
         }
+
+        public async Task<IEnumerable<string>> GetEnrolleeEmails(BulkEmailType bulkEmailType)
+        {
+            Expression<Func<Enrollee, bool>> predicate = bulkEmailType switch
+            {
+                BulkEmailType.CommunityPractice => e => e.EnrolleeCareSettings.Any(cs => cs.CareSettingCode == (int)CareSettingType.CommunityPractice),
+                BulkEmailType.CommunityPharmacy => e => e.EnrolleeCareSettings.Any(cs => cs.CareSettingCode == (int)CareSettingType.CommunityPharmacy),
+                BulkEmailType.HealthAuthority => e => e.EnrolleeCareSettings.Any(cs => cs.CareSettingCode == (int)CareSettingType.CommunityPharmacy),
```
Comment (johnnyl6) [inline prime-dotnet-webapi/Services/EnrolleeService.cs:914]: Should it be `cs => cs.CareSettingCode == (int)CareSettingType.HealthAuthority` ?

**Review by johnnyl6 — APPROVED** [review]
LGTM🎉

**Pull request merged**